lib/perso.ml

```diff
@@ -1216,6 +1247,302 @@ let warning_use_has_parents_before_parent (fname, bp, ep) var r =
   |> !GWPARAM.syslog `LOG_WARNING;
   r
 
+let cousins_t = ref None
+let cousins_dates_t = ref None
+
+let max_l1_l2_aux env =
+  let max_l1 = match get_env "max_anc_level" env with Vint i -> i | _ -> 5 in
+  let max_l2 = match get_env "max_desc_level" env with Vint i -> i | _ -> 5 in
+  (max_l1 - 1, (2 * max_l1) + max_l2 + 20)
+
+let get_min_max_dates conf base l =
+  let dates =
+    let rec loop acc = function
+      | [] -> acc
+      | (ip, _, _, _) :: l -> (
+          let not_dead = get_death (poi base ip) = NotDead in
+          let birth_date, death_date, _ =
+            Gutil.get_birth_death_date (poi base ip)
+          in
+          match (birth_date, death_date) with
```

Comment:
maybe try to match on only one at a time so it is more readable.
Also you can use `Date.cdate_to_dmy_opt` to simplify the match

Reply:
a comment on what `get_min_max_dates` does would be welcome too :+1: